src/Draco.Compiler/Internal/Syntax/Formatting/DracoFormatter.cs

```diff
@@ -0,0 +1,404 @@
+            {
+                this.formatter.CurrentToken.Text = node.Text + " " + comment;
+                this.formatter.NextToken.DoesReturnLine = true;
+            }
+        }
+        var leadingComments = node.LeadingTrivia
+            .Where(x => x.Kind == TriviaKind.LineComment || x.Kind == TriviaKind.DocumentationComment)
+            .Select(x => x.Text)
+            .ToArray();
+        this.formatter.CurrentToken.LeadingTrivia ??= [];
+        this.formatter.CurrentToken.LeadingTrivia.AddRange(leadingComments);
+        if (this.formatter.CurrentToken.LeadingTrivia.Count > 0)
+        {
+            this.formatter.CurrentToken.DoesReturnLine = true;
+        }
+    }
+
+    public override void VisitSeparatedSyntaxList<TNode>(Api.Syntax.SeparatedSyntaxList<TNode> node)
+    {
+        if (node is Api.Syntax.SeparatedSyntaxList<Api.Syntax.ParameterSyntax>
+            || node is Api.Syntax.SeparatedSyntaxList<Api.Syntax.ExpressionSyntax>)
+        {
+            this.formatter.CreateMaterializableScope(this.settings.Indentation,
+                FoldPriority.AsSoonAsPossible,
+                () => base.VisitSeparatedSyntaxList(node)
+            );
+        }
+        else
+        {
+            base.VisitSeparatedSyntaxList(node);
+        }
+    }
+
+    public override void VisitParameter(Api.Syntax.ParameterSyntax node)
+    {
+        this.formatter.CurrentToken.DoesReturnLine = this.formatter.Scope.IsMaterialized;
+        base.VisitParameter(node);
+    }
+
+    public override void VisitDeclaration(Api.Syntax.DeclarationSyntax node)
+    {
+        this.formatter.CurrentToken.DoesReturnLine = true;
+        var type = node.GetType();
+        this.formatter.OnDifferent(node switch
+        {
+            Api.Syntax.FunctionDeclarationSyntax _ => node, // always different, that what we want.
+            _ => type,
+        }, previous =>
+        {
+            if (previous == null) return;
+            this.formatter.CurrentToken.LeadingTrivia = [""]; // a newline is created between each leading trivia.
+        });
+        base.VisitDeclaration(node);
+    }
+
+    public override void VisitStringExpression(Api.Syntax.StringExpressionSyntax node)
+    {
+        if (node.OpenQuotes.Kind != TokenKind.MultiLineStringStart)
+        {
+            node.OpenQuotes.Accept(this);
+            foreach (var item in node.Parts.Tokens)
+            {
+                this.formatter.CurrentToken.DoesReturnLine = false;
+                item.Accept(this);
+            }
+            this.formatter.CurrentToken.DoesReturnLine = false;
+            node.CloseQuotes.Accept(this);
+            return;
+        }
+        node.OpenQuotes.Accept(this);
+        using var _ = this.formatter.CreateScope(this.settings.Indentation);
+        var blockCurrentIndentCount = SyntaxFacts.ComputeCutoff(node).Length;
+
+        var i = 0;
+        var shouldIndent = true;
+        for (; i < node.Parts.Count; i++)
```

Comment:
```suggestion
        var shouldIndent = true;
        for (var i = 0;; i < node.Parts.Count; i++)
```

Reply:
That's because previously i was used in the outer context, but its' not the case anymore, good catch.